### PR TITLE
use pipenv for python environment management

### PR DESCRIPTION
**Pull request by tkellen**
This aims to streamline the workflow of test generation through
the usage of https://pipenv.readthedocs.io/en/latest/

Contributes to gh-1691

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -320,15 +320,15 @@ any other valid frontmatter field | see the frontmatter definitions.
 
 Generated files are managed using the `make.py` Python script located in the root of this repository. To use it, first install the required Python packages via the following command:
 
-    pip install --requirement tools/generation/requirements.txt
+    pip install pipenv && pipenv install
```
Comment (chicoxyzzy) [inline CONTRIBUTING.md]: Possibly it'll be useful to add a note that to use `pipenv`, Python should be in PATH

**Comment by chicoxyzzy** [comment]
`pipenv` doesn't work for me for some reason
```bash
➜  test262 git:(pipenv) pipenv install
Warning: the environment variable LANG is not set!
We recommend setting this in ~/.profile (or equivalent) for proper expected behavior.
Creating a virtualenv for this project...
Pipfile: /Users/chicoxyzzy/Projects/Personal/test262/Pipfile
Using /usr/local/bin/python2.7 (2.7.15) to create virtualenv...
⠋Traceback (most recent call last):
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/runpy.py", line 162, in _run_module_as_main
    "__main__", fname, loader, pkg_name)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/runpy.py", line 72, in _run_code
    exec code in run_globals
  File "/Users/chicoxyzzy/Library/Python/2.7/lib/python/site-packages/pipenv/pew/__main__.py", line 8, in <module>
    import pew
  File "/Users/chicoxyzzy/Library/Python/2.7/lib/python/site-packages/pipenv/patched/pew/__init__.py", line 3, in <module>
    from . import pew
  File "/Users/chicoxyzzy/Library/Python/2.7/lib/python/site-packages/pipenv/patched/pew/pew.py", line 44, in <module>
    from pew._utils import (check_call, invoke, expandpath, own, env_bin_dir,
  File "/Users/chicoxyzzy/Library/Python/2.7/lib/python/site-packages/pipenv/patched/pew/_utils.py", line 25, in <module>
    encoding = locale.getlocale()[1] or 'ascii'
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/locale.py", line 562, in getlocale
    return _parse_localename(localename)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/locale.py", line 475, in _parse_localename
    raise ValueError, 'unknown locale: %s' % localename
ValueError: unknown locale: UTF-8

Virtualenv location:
Traceback (most recent call last):
  File "/Users/chicoxyzzy/Library/Python/2.7/bin/pipenv", line 11, in <module>
    sys.exit(cli())
  File "/Users/chicoxyzzy/Library/Python/2.7/lib/python/site-packages/pipenv/vendor/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/Users/chicoxyzzy/Library/Python/2.7/lib/python/site-packages/pipenv/vendor/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/Users/chicoxyzzy/Library/Python/2.7/lib/python/site-packages/pipenv/vendor/click/core.py", line 1066, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/chicoxyzzy/Library/Python/2.7/lib/python/site-packages/pipenv/vendor/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Users/chicoxyzzy/Library/Python/2.7/lib/python/site-packages/pipenv/vendor/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/Users/chicoxyzzy/Library/Python/2.7/lib/python/site-packages/pipenv/cli.py", line 435, in install
    selective_upgrade=selective_upgrade,
  File "/Users/chicoxyzzy/Library/Python/2.7/lib/python/site-packages/pipenv/core.py", line 1759, in do_install
    pypi_mirror=pypi_mirror,
  File "/Users/chicoxyzzy/Library/Python/2.7/lib/python/site-packages/pipenv/core.py", line 614, in ensure_project
    path_to_python = which('python') or which('py')
  File "/Users/chicoxyzzy/Library/Python/2.7/lib/python/site-packages/pipenv/core.py", line 115, in which
    p = os.path.join(location, 'bin', command)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/posixpath.py", line 70, in join
    elif path == '' or path.endswith('/'):
AttributeError: 'NoneType' object has no attribute 'endswith'
/Users/chicoxyzzy/Library/Python/2.7/lib/python/site-packages/pipenv/_compat.py:113: ResourceWarning: Implicitly cleaning up <TemporaryDirectory '/var/folders/kv/lkrh789n7wb_47f57lnjbk300000gn/T/pipenv-Se8GV6-requirements'>
  warnings.warn(warn_message, ResourceWarning)
```

**Comment by chicoxyzzy** [comment]
This solved my issue with UTF-8
https://coderwall.com/p/-k_93g/mac-os-x-valueerror-unknown-locale-utf-8-in-python

Now I have another error
```
➜  test262 git:(pipenv) pipenv install
Creating a virtualenv for this project...
Pipfile: /Users/chicoxyzzy/Projects/Personal/test262/Pipfile
Using /usr/local/bin/python2.7 (2.7.15) to create virtualenv...
⠋Running virtualenv with interpreter /usr/local/bin/python2.7
Traceback (most recent call last):
  File "/Users/chicoxyzzy/Library/Python/2.7/lib/python/site-packages/virtualenv.py", line 22, in <module>
    import zlib
ImportError: No module named zlib
Error while trying to remove the /Users/chicoxyzzy/.local/share/virtualenvs/test262-V_87uGqr env:
No such file or directory

Virtualenv location:
Traceback (most recent call last):
  File "/Users/chicoxyzzy/Library/Python/2.7/bin/pipenv", line 11, in <module>
    sys.exit(cli())
  File "/Users/chicoxyzzy/Library/Python/2.7/lib/python/site-packages/pipenv/vendor/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/Users/chicoxyzzy/Library/Python/2.7/lib/python/site-packages/pipenv/vendor/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/Users/chicoxyzzy/Library/Python/2.7/lib/python/site-packages/pipenv/vendor/click/core.py", line 1066, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/chicoxyzzy/Library/Python/2.7/lib/python/site-packages/pipenv/vendor/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Users/chicoxyzzy/Library/Python/2.7/lib/python/site-packages/pipenv/vendor/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/Users/chicoxyzzy/Library/Python/2.7/lib/python/site-packages/pipenv/cli.py", line 435, in install
    selective_upgrade=selective_upgrade,
  File "/Users/chicoxyzzy/Library/Python/2.7/lib/python/site-packages/pipenv/core.py", line 1759, in do_install
    pypi_mirror=pypi_mirror,
  File "/Users/chicoxyzzy/Library/Python/2.7/lib/python/site-packages/pipenv/core.py", line 614, in ensure_project
    path_to_python = which('python') or which('py')
  File "/Users/chicoxyzzy/Library/Python/2.7/lib/python/site-packages/pipenv/core.py", line 115, in which
    p = os.path.join(location, 'bin', command)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/posixpath.py", line 70, in join
    elif path == '' or path.endswith('/'):
AttributeError: 'NoneType' object has no attribute 'endswith'
/Users/chicoxyzzy/Library/Python/2.7/lib/python/site-packages/pipenv/_compat.py:113: ResourceWarning: Implicitly cleaning up <TemporaryDirectory '/var/folders/kv/lkrh789n7wb_47f57lnjbk300000gn/T/pipenv-4dvBLi-requirements'>
  warnings.warn(warn_message, ResourceWarning)
```

**Comment by chicoxyzzy** [comment]
`brew reinstall python2` solved an issue

**Comment by chicoxyzzy** [comment]
I really think that this PR should be merged. It simplifies things a lot.

**Comment by tkellen** [comment]
Glad it was helpful @chicoxyzzy.

It's a bit unclear to me who would make a determination about this so I am going with the top three committers. Any thoughts here @leobalter @rwaldron @jugglinmike?

File: CONTRIBUTING.md (hunk 1 above)
Comment (tkellen) [reply CONTRIBUTING.md]: If this patch is desirable to the maintainers of this project I'd be happy to add some prose to that effect.

**Comment by leobalter** [comment]
I'm neutral to this change and waiting for a feedback from @jugglinmike. I already asked him to get some eyes in here.

@tkellen meanwhile, can we get the CI green?

**Comment by leobalter** [comment]
After reading https://github.com/tc39/test262/issues/1691#issuecomment-417953973

> For now, I'd like to see how much farther we can get with better documentation.
That includes specifying the expected Python version (which we really ought to
do in any event), and using a slightly different installation command that
should prevent the original problem.

I like the simplicity of #1703 keeping a minimal set of requirements and improving the docs.

IMO, I'd like to follow @jugglinmike's recommendation and see how it goes. I'd prefer to avoid the extra complexity while it's possible.

We can eventually work back on this if necessary, but I'm closing for now. Thanks for all the work, I definitely learned some interesting approaches with Python.

**Pull request closed without merging**